### PR TITLE
Get rid of Firebase Crashlytics, move to MSAppCenter Crashlytics & Analytics #infra

### PR DESCRIPTION
**Pull request by Kaspik**
## Changes:
- remove CLS_LOG and these logs
- move to MSAppCenter logging out of Firebase Crashlytics
- update OCMock framework for Unit Tests to new M1 compatible

## Tested:
- Processors:
  - [ ] Intel
  - [x] Apple Silicon
- macOS Versions:
  - [ ] 10.12.x (Sierra)
  - [ ] 10.13.x (High Sierra)
  - [ ] 10.14.x (Mojave)
  - [ ] 10.15.x (Catalina)
  - [x] 11.x (Big Sur)
- Xcode Version: 12.3
- Localizations:
  - [x] English
  - [ ] Spanish
  - [ ] Other (please specify)

### CODE REVIEW

**Comment by Kaspik** [comment]
Ignore during review:
- OCMock changes (in Frameworks folder)
- most of the code changes except SPAppController

It's just removal of a lot of logs for now - we will get them back in more usable way in upcoming PRs. Merging, but still feel free to review and check please. :)

File: Frameworks/OCMock.framework/Modules [hunk 1]
```diff
@@ -0,0 +1 @@
+Versions/Current/Modules
```
Comment (Jason-Morcos) [inline Frameworks/OCMock.framework/Modules:1]: Could we not get this framework through the swift package manager? I worry about static dependencies as they have a way of getting out of date

File: Source/Controllers/DataControllers/SPDatabaseStructure.m [hunk 2]
```diff
@@ -314,10 +314,7 @@ - (void)queryDbStructureWithUserInfo:(NSDictionary *)userInfo
 		 }
 
             NSString *theQuery = [NSString stringWithFormat:@"SHOW FULL COLUMNS FROM %@ FROM %@", [aTableName backtickQuotedString], [currentDatabase backtickQuotedString]];
-
-            CLS_LOG(@"theQuery: %@", theQuery);
             SPLog(@"theQuery: %@", theQuery);
-            [[FIRCrashlytics crashlytics] setCustomValue:mySQLConnection.serverVersionString forKey:@"serverVersion"];
```
Comment (Jason-Morcos) [inline Source/Controllers/DataControllers/SPDatabaseStructure.m:320]: Can we do a similar thing with MSAppCenter? To set specific analytics params

Comment (Kaspik) [reply Source/Controllers/DataControllers/SPDatabaseStructure.m:320]: We can have attachments when pre-processing crashes, but I've not yet implemented them

File: Frameworks/OCMock.framework/Modules (hunk 1 above)
Comment (Kaspik) [reply Frameworks/OCMock.framework/Modules:1]: Nope, doesn't support ... but it's only for tests so the reason I updated it is the new M1 support, and we won't need it for years again 😂

File: Source/Controllers/DataControllers/SPDatabaseStructure.m (hunk 2 above)
Comment (jamesstout) [reply Source/Controllers/DataControllers/SPDatabaseStructure.m:320]: You can add [custom events](https://docs.microsoft.com/en-us/appcenter/sdk/analytics/ios):

```objc
[MSACAnalytics trackEvent:@"something clicked"];
```
or 
```objc
NSDictionary *properties = @{@"Category" : @"LoadData", @"TimeTaken" : _warning._time2};
[MSACAnalytics trackEvent:@"Init LoadData" withProperties: properties];
```

You can pause and resume, when network bandwidth is needed, e.g. exporting a big table"
```objc
[MSACAnalytics pause];
[MSACAnalytics resume];
```
But this is for analytics, not crash reports.

File: Source/Controllers/SPAppController.m [hunk 3]
```diff
@@ -247,28 +245,6 @@ - (void)applicationDidFinishLaunching:(NSNotification *)notification
...
 
-    executeOnBackgroundThread(^{
-        NSUserDefaults *prefs = [NSUserDefaults standardUserDefaults];
-        FIRCrashlytics *crashlytics = FIRCrashlytics.crashlytics;
-
-        // fake the dbViewInfoPanelSplit being open
-        NSMutableArray *dbViewInfoPanelSplit = [[NSMutableArray alloc] initWithCapacity:2];
-        [dbViewInfoPanelSplit addObject:@"0.000000, 0.000000, 359.500000, 577.500000, NO, NO"];
-        [dbViewInfoPanelSplit addObject:@"0.000000, 586.500000, 359.500000, 190.500000, NO, NO"];
-        [prefs setObject:dbViewInfoPanelSplit forKey:@"NSSplitView Subview Frames DbViewInfoPanelSplit"];
```
Comment (Jason-Morcos) [inline Source/Controllers/SPAppController.m:258]: @Kaspik Was this an intentional remove? I think these lines are unrelated to crashlytics

File: Source/Controllers/SPAppController.m [hunk 4]
```diff
@@ -190,16 +194,10 @@ - (void)awakeFromNib
...
-    // default is FIRLoggerLevelNotice, and for App Store apps
-    // cannot be set higher than FIRLoggerLevelNotice
-    [[FIRConfiguration sharedInstance] setLoggerLevel:FIRLoggerLevelDebug];
-#endif
+- (void)applicationDidFinishLaunching:(NSNotification *)notification {
+    // Use 30 MB for storage for logs
+    [MSACAppCenter setMaxStorageSize:(30 * 1024 * 1024) completionHandler:nil];
+    [MSACAppCenter start:@"65535bfb-1763-40fd-896b-a3aaae06227f" withServices:@[[MSACAnalytics class], [MSACCrashes class]]];
 
     // has to be after FIRApp configure
```
Comment (Jason-Morcos) [inline Source/Controllers/SPAppController.m:202]: Can remove this comment

File: Source/Other/Data/SPConstants.h [hunk 5]
```diff
@@ -763,23 +761,3 @@ static inline __attribute__((always_inline)) void SABenchmark(void (^block)(void
...
-          foreachEnum; \
-          foreachEnum = nil ) \
-        for ( IMP foreachNext = [foreachEnum methodForSelector:@selector(nextObject)]; \
-              foreachNext; \
-              foreachNext = NULL ) \
-            for ( type object = foreachNext(foreachEnum, @selector(nextObject)); \
-                  object; \
-                  object = foreachNext(foreachEnum, @selector(nextObject)) )
-
-#define foreach(object, collection) foreacht(id, object, (collection))
```
Comment (Jason-Morcos) [inline Source/Other/Data/SPConstants.h:785]: Does this removal pose any risk of breaking things?

File: sequel-ace.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -2914,31 +2903,6 @@
 /* End PBXResourcesBuildPhase section */
 
 /* Begin PBXShellScriptBuildPhase section */
-		1A47995F2543320B00D29E6E /* Run Crashlytics */ = {
```
Comment (Jason-Morcos) [inline sequel-ace.xcodeproj/project.pbxproj:2917]: Does MSAppCenter automatically upload DSYMs? Is there some step we need to take to upload them instead?

File: Source/Controllers/SPAppController.m (hunk 3 above)
Comment (Kaspik) [reply Source/Controllers/SPAppController.m:258]: Reverted

File: Source/Other/Data/SPConstants.h (hunk 5 above)
Comment (Kaspik) [reply Source/Other/Data/SPConstants.h:785]: Shouldn't it was unused.

File: sequel-ace.xcodeproj/project.pbxproj (hunk 6 above)
Comment (Kaspik) [reply sequel-ace.xcodeproj/project.pbxproj:2917]: They have CLi you can install to terminal, but easiest is just drop the zip to the UI

File: Source/Other/Data/SPConstants.h (hunk 5 above)
Comment (jamesstout) [reply Source/Other/Data/SPConstants.h:785]: Nope, was just for my tests, @Kaspik ~~deleted~~ sorted them.

File: sequel-ace.xcodeproj/project.pbxproj (hunk 6 above)
Comment (jamesstout) [reply sequel-ace.xcodeproj/project.pbxproj:2917]: Ah no .. we need a script. I have one....

File: Source/Controllers/SPAppController.m (hunk 3 above)
Comment (jamesstout) [reply Source/Controllers/SPAppController.m:258]: No we need that. That was the 'fix' for the eye-information going whacky bug.

Comment (Jason-Morcos) [reply Source/Controllers/SPAppController.m:258]: Looks like @Kaspik reverted these lines back in main. Resolving convo! :)
